src/kOS.Safe/Serialization/JSONFormatter.cs

```diff
@@ -1,15 +1,20 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-
+﻿using System.Collections.Generic;
 
 namespace kOS.Safe.Serialization
 {
-    public class JSONFormatter : Formatter
+    public class JsonFormatter : IFormatWriter, IFormatReader
     {
-        private static readonly JSONFormatter instance = new JSONFormatter();
+        private static readonly JsonFormatter instance;
+
+        public static IFormatReader ReaderInstance
+        {
+            get 
+            {
+                return instance; 
+            }
+        }
 
-        public static JSONFormatter Instance
+        public static IFormatWriter WriterInstance
```

Comment:
I do not love the singleton pattern mostly because i think it makes testing consuming classes harder. I made this change based on splitting up the formatter instance. I could be convinced that having two 'instance' accessors is not optimal